annotations/tekton-annotations/src/main/java/io/dekorate/tekton/generator/TektonApplicationGenerator.java

```diff
@@ -0,0 +1,93 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.dekorate.tekton.generator;
+
+import java.lang.annotation.Annotation;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.lang.model.element.Element;
+
+import io.dekorate.BuildService;
+import io.dekorate.BuildServiceFactories;
+import io.dekorate.DekorateException;
+import io.dekorate.Generator;
+import io.dekorate.Session;
+import io.dekorate.SessionListener;
+import io.dekorate.WithProject;
+import io.dekorate.WithSession;
+import io.dekorate.config.ConfigurationSupplier;
+import io.dekorate.deps.kubernetes.api.model.KubernetesList;
+import io.dekorate.hook.ImageBuildHook;
+import io.dekorate.tekton.adapter.TektonConfigAdapter;
+import io.dekorate.tekton.annotation.TektonApplication;
+import io.dekorate.tekton.config.TektonConfig;
+import io.dekorate.tekton.config.TektonConfigCustomAdapter;
+import io.dekorate.tekton.handler.TektonHandler;
+import io.dekorate.kubernetes.config.ImageConfiguration;
+import io.dekorate.kubernetes.configurator.ApplyBuildToImageConfiguration;
+import io.dekorate.project.ApplyProjectInfo;
+import io.dekorate.project.Project;
+
+public interface TektonApplicationGenerator extends Generator, WithSession, WithProject, SessionListener {
+
+  String TEKTON = "tekton";
+
+  default String getKey() {
+    return TEKTON;
+  }
+
+  default Class<? extends Annotation> getAnnotation() {
+    return TektonApplication.class;
+  }
+
+
+  default void add(Element element) {
+    TektonApplication tektonApplication = element.getAnnotation(TektonApplication.class);
+    TektonConfig tektonConfig = TektonConfigCustomAdapter.newBuilder(getProject(), tektonApplication).build();
+
+    on(new ConfigurationSupplier<>(TektonConfigAdapter.newBuilder(element.getAnnotation(TektonApplication.class))
+                                   .accept(new ApplyBuildToImageConfiguration())
+                                   .accept(new ApplyProjectInfo(getProject()))));
+  }
+
+  default void add(Map map) {
+    TektonConfig tektonConfig = TektonConfigAdapter.newBuilder((Map) map.get(TektonApplication.class.getName())).build();
+    on(new ConfigurationSupplier<>(TektonConfigAdapter.newBuilder(propertiesMap(map, TektonApplication.class))
+                                   .accept(new ApplyBuildToImageConfiguration())
+                                   .accept(new ApplyProjectInfo(getProject()))));
+  }
+
+    default void on(ConfigurationSupplier<TektonConfig> config) {
+      Session session = getSession();
+      session.configurators().add(config);
+      session.handlers().add(new TektonHandler(session.resources(), session.configurators()));
+      session.addListener(this);
+  }
+
+  default void onClosed() {
+    //We ned to set the TTCL, becuase the KubenretesClient used in this part of code, needs TTCL so that java.util.ServiceLoader can work.
```

Comment:
```suggestion
    //We need to set the TCCL, becuase the KubernetesClient used in this part of code, needs TCCL so that java.util.ServiceLoader can work.
```